app/lib/features/pins/pin_utils/pin_utils.dart

```diff
@@ -13,95 +13,102 @@ import 'package:mime/mime.dart';
 typedef PinAttachment = ({AttachmentType type, File file});
 
 class PinUtils {
+  // belongs separately, useful for showing file picker directly (desktop)
+  static void _onTapFileSelection(WidgetRef ref) async {
+    FilePickerResult? result = await FilePicker.platform.pickFiles(
+      type: FileType.any,
+      allowMultiple: true,
+    );
+    if (result != null) {
+      List<PinAttachment> newAttachments =
+          result.paths.map<PinAttachment>((path) {
+        var file = File(path!);
+        var pinAttachment = (type: AttachmentType.file, file: file);
+        return pinAttachment;
+      }).toList();
+
+      List<PinAttachment> attachments =
+          ref.read(selectedPinAttachmentsProvider);
+      var attachmentNotifier =
+          ref.read(selectedPinAttachmentsProvider.notifier);
+      if (attachments.isNotEmpty) {
+        attachments.addAll(newAttachments);
+        attachmentNotifier.update((state) => [...attachments]);
+      } else {
+        attachmentNotifier.update((state) => newAttachments);
+      }
+    }
+  }
+
   // attachment selection sheet
   static void showAttachmentSelection(
     BuildContext context,
     WidgetRef ref,
   ) async {
-    await showModalBottomSheet<void>(
-      isDismissible: true,
-      context: context,
-      builder: (context) => AttachmentOptions(
-        onTapCamera: () async {
-          XFile? imageFile =
-              await ImagePicker().pickImage(source: ImageSource.camera);
-          if (imageFile != null) {
-            File file = File(imageFile.path);
-            List<PinAttachment> attachmentList = [
-              (type: AttachmentType.camera, file: file),
-            ];
-            ref
-                .read(selectedPinAttachmentsProvider.notifier)
-                .update((state) => [...attachmentList]);
-          }
-        },
-        onTapImage: () async {
-          List<XFile> imageFiles = await ImagePicker().pickMultiImage();
-          List<PinAttachment> newAttachments = [];
+    isLargeScreen(context)
```

Comment:
Why we have applied this condition here for the desktop?

Reply:
Over desktop, there are only files selection not custom image, video type selection. This change will directly open ```file_picker``` instead of `image_picker``` and also we don't need to show bottom sheet file selection UI here.

Reply:
> Over desktop, there are only files selection not custom image, video type selection. This change will directly open file_picker instead of `image_picker```

Separate options allows users to select file specific extension only.

> and also we don't need to show bottom sheet file selection UI here.

Bottom sheet UI was not looking good just because of side sheet concept, if we have full page screen then I think it will not create any issue in UI.